src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/services/TestCaseDisplayService.kt

```diff
@@ -263,36 +221,23 @@ class TestCaseDisplayService(private val project: Project) {
             showLibraryContentsField.isAccessible = true
             showLibraryContentsField.set(chooser, false)
         } catch (_: Exception) {
-            // could not set field
-            // ignoring the exception is acceptable as this part is not critical
+            // Could not set field
+            // Ignoring the exception is acceptable as this part is not critical
         }
-
         chooser.showDialog()
 
-        // get selected class or return if no class was selected
+        // Get the selected class or return if no class was selected
         val selectedClass = chooser.selected ?: return
 
-        // insert test case components into selected class
+        // Insert test case components into selected class
         appendTestsToClass(testCaseComponents, selectedClass)
 
-        // schedule telemetry
-        val telemetryService = ApplicationManager.getApplication().getService(TestGenieTelemetryService::class.java)
-        telemetryService.scheduleTestCasesForTelemetry(
-            selectedTestCases.map {
-                val modified = (testCasePanels[it]!!.getComponent(1) as EditorTextField).text
-                val original = originalTestCases[it]!!
-
-                TestGenieTelemetryService.ModifiedTestCase(
-                    original,
-                    modified
-                )
-            }.filter {
-                it.modified != it.original
-            }
-        )
-
         // The scheduled tests will be submitted in the background
         // (they will be checked every 5 minutes and also when the project is closed)
+        scheduleTelemetry(selectedTestCases)
+
+        // Remove the selected test cases from the cache and the tool window UI
+        removeTestCases(selectedTestCasePanels)
```

Comment:
So, now you remove at the latest possible moment. Is that what allowed you to avoid `null`pointers that we talked about?

Reply:
Yes, exactly. I did not remove the test panels from `allTestCases` i think